Categories.txt improved Dutch and German translations

demasterr opened the pull request:
Improved the translations for Dutch and German by adding new translations, fixing some wrong translations, and reordering some of the translations.

*Below the most important remarks of changes are listed. The changes with warning signs may need an extra review or contain a question.*

A few remarks for the Dutch translations:
- The earlier  used 'autodelen' for 'car sharing' is probably not the most common phrase in Dutch. 'Carpool' is a better translation, as this is the known concept for sharing your car.
- Hospital received an extra synonym for Belgian people. In the Netherlands (Holland) this word is not used, however, Belgian people may use 'hospitaal' as a synonym for 'hospital'.
- The translation of 'Dumpster|trash' was now translated in Dutch as 'uitschot'. This word is more used for calling a person trash, not so much for throwing away stuff. We don't want people to get the impressions that there are some trash people at a location ;)
- For hospital there was the Dutch translation 'tandarts', which is a 'dentist'. This probably shouldn't belong here.
- ⚠️ 'County' was translated as 'graafschap' for Dutch. This is a very old translation for it and will not be used in normal day usage. 'Pronvincie' is a better translation here. The word may be used differently in other countries though, which would mean that this translation may then be wrong.
- Island was translated in Dutch as 'eilandje', which would mean that it is only used for small islands. 'Eiland' is the right translation for all sizes.
- ⚠️ Racetrack was changed to 'renbaan' for Dutch. This is a better translation UNLESS racetrack is also used for cars. 'Renbaan' is only used for animal racing. So the question is, is racetrack used for animals only or also for cars?
- ⚠️ The German translation of Hvac|Air conditioner seems a bit off. Heizung translates to heater. Air conditioner directly translated gives Klimaanlage. However, I'm not German myself so I cannot fully guarantee that this is the right translation for this point.
- Nursery|child care was in German translated to Kinderzimmer, which is a child's room
- A defibrillator is in Dutch called an AED. I don't expect most Dutch people to know the word 'defibrillator'. Though, defibrillator is an accepted Dutch word, so it is still valid here.
- ⚠️ Variety Store did not have a Spanish translation. It now contained `<Todo a cien>`. I replaced it with the Google Translate suggestion. However, this translation could be wrong! Anyhow, it can never be any worse than seeing `<Todo a cien>` in your GUI.
EDIT: at least one Spanish person confirmed this translation for me.

Could someone explain the usage of `place-locality`? I believe the Dutch translation is wrong here. However, I do think the current translation is wrong.

mapsmetest commented:
Can one of the admins verify this patch?

HenkGrent commented on data/categories.txt:
```diff
@@ -3825,7 +3831,7 @@ amenity-toilets|@toilet
 ar:^حمام|مرحاض
 cs:wc|toalety
-nl:wc
+nl:WC
```
Consider adding "Toilet"

demasterr replied:
Good one, will add!

mpimenov commented:
Hello. Please fill in the CLA: https://maps.me/cla/

demasterr commented:
Done!

casperboone commented on data/categories.txt:
```diff
@@ -7447,7 +7455,7 @@ ru:^Карта
 cs:^Turistická mapa
 da:^Turistkort
-nl:^Toeristische kaart
+nl:^Toeristische kaart|toeristen kaart
```
I think this should be 'toeristenkaart'

demasterr replied:
Fixed!

casperboone commented on data/categories.txt:
```diff
@@ -7254,10 +7262,10 @@ ru:^Рыбный магазин|рыба|морепродукты|рыбная 
 cs:^Prodej ryb
 da:^Fiskehandler
-nl:^Visboer
+nl:^Visboer|vis markt|vis|zeevoedsel|schelpdier
```
I think this should be 'vismarkt'

demasterr replied:
Fixed!

casperboone commented on data/categories.txt:
```diff
@@ -6612,7 +6620,7 @@ ru:^Телефон для экстренных вызовов
 cs:^Tísňového volání
 da:^Nødtelefon
-nl:^Praatpaal
+nl:^Praatpaal|4nood telefoon
```
I think this should be 'noodtelefoon'

demasterr replied:
Yes, fixed!

casperboone commented on data/categories.txt:
```diff
@@ -6420,7 +6428,7 @@ ru:^Автомойка
 cs:^Myčka aut
 da:^Bilvask
-nl:^Car wash
+nl:^Car wash|auto was
```
I think 'autowasstraat' or 'wasstraat' are more appropriate translations

demasterr replied:
Yes auto was is probably more slang. I'll take your suggestion 👍

casperboone commented on data/categories.txt:
```diff
@@ -6231,6 +6238,7 @@ ru:^Копировальный центр|печать|полиграфия
 ar:^محل نسخ|طباعة
 cs:^Kopírovací obchod|Tiskárny
 da:^Kopieringsbutik|Trykkeri
+nl:^Copy shop|Kopieer winkel|printer|drukker
```
I think this should be 'kopieerwinkel'

demasterr replied:
Fixed!

casperboone commented on data/categories.txt:
```diff
@@ -5407,7 +5414,7 @@ ru:4^Кемпинг
 cs:4^Kempování|camping|kemp|camp
 da:^Camping|campingplads|teltplads|lejrplads
-nl:4^Camping|kamperen
+nl:4^Camping|kamperen|kampeer plek
```
I think this should be 'kampeerplek'

demasterr replied:
Fixed!

casperboone commented on data/categories.txt:
```diff
@@ -3629,7 +3635,7 @@ ru:^Мусорный контейнер|мусорка
 cs:^Odpadky|koš|popelnice|uložiště odpadu
 da:^Bortskaffelse af affald|skraldespand|skrald
-nl:^Uitschot|afval|prullenbak|vuilnis
+nl:^Afval container|afval|prullenbak|vuilnis
```
I think this should be 'afval container'

demasterr replied:
Thank you!

demasterr commented:
@mpimenov The CLA was signed. Could the PR be reviewed?

mpimenov commented on data/categories.txt:
```diff
@@ -2189,7 +2189,7 @@ ru:^Место для пикника|3Пикник
 cs:3^Piknik|picnic|picnik
 da:3^Picnic
-nl:3^Picnicplaats|3picnicweide|3picnictafel
+nl:3^Picknickplaats|3picknickweide|3picknicktafel
```
Are you sure all three synonyms need the "3" prefix? Do we need to suggest all three of them after typing 3 letters?

demasterr replied:
You are right, only the first should be suggested. Fixed!

mpimenov commented on data/categories.txt:
```diff
@@ -5471,7 +5478,7 @@ ru:^Офис|компания|контора|фирма|бизнес|бюро|у
 cs:^Kancelář
 da:^Kontor|firma|virksomhed
-nl:^Kantoor|ambt
+nl:^Kantoor|ambt|bedrijf|bedrijspand|bedrijfskantoor
```
s/bedrijspand/bedrijfspand/ ?

demasterr replied:
Fixed!

mpimenov commented on data/categories.txt:
```diff
@@ -6073,7 +6080,7 @@ da:WiFi|Wi-Fi
 fi:WiFi|Wi-Fi
 fr:WiFi|Wi-Fi
-de:WLAN|WiFi|Wi-Fi
+de:WiFi|Wi-Fi|WLAN
```
nit: the order doesn't matter here

demasterr replied:
Alright, in that case, I'll revert this change.

mpimenov commented:
@demasterr I can't see you in our CLA list. Are you sure you have used this github account name?

mpimenov commented:
Regarding the circumflex: sorry for the confusion, the correct way to make a translation to be shown in the UI is now to edit https://github.com/mapsme/omim/blob/master/data/strings/types_strings.txt

demasterr commented:
@mpimenov I signed under the GitHub Account `demasterr`. I've signed it again to be sure.

demasterr commented:
Is categories.txt not used any more? Otherwise it may be wise to delete the file from the master.

I'll have a look if I can move the changes suggested in this PR to https://github.com/mapsme/omim/blob/master/data/strings/types_strings.txt somewhere this week.

mpimenov commented:
Also, note the all strings shown in the UI should be in this directory, should you want to fix them too: https://github.com/mapsme/omim/blob/master/data/strings

mpimenov commented:
Please squash all the commits into one.

mpimenov commented:
We'll look into making it more clear, but for now:
* Strings from the strings/ directory are shown in the UI.
* Strings from the categories* files are used as synonyms for search.

mpimenov commented:
> I'll have a look if I can move the changes suggested in this PR to https://github.com/mapsme/omim/blob/master/data/strings/types_strings.txt somewhere this week.

The changes here are fine but you may want to repeat some of them there too.

demasterr commented:
@mpimenov 
> Please squash all the commits into one.

I've no idea what I was doing now. Could you advise me on how to do this?

mpimenov commented:
git rebase -i
(interactive rebase)

demasterr commented:
@mpimenov It took some time but I finally managed to do it. Missed the `-f` with the push 🤦‍♂️

demasterr commented:
I also found one typo while doing the rebase. I had `3Car poolen`. I changed it to `Carpoolen`. The suggestion after 3 characters was also removed, as it was not needed.

mpimenov commented:
Please change this large commit message to a shorter "Categories.txt improved Dutch and German translations" or similar.

demasterr commented:
@mpimenov This should be better 👍
